components/lib/crash/src/test/java/mozilla/components/lib/crash/BreadcrumbTest.kt

```diff
@@ -163,13 +163,121 @@ class BreadcrumbTest {
         sleep(100) /* make sure time elapsed */
         val afterDate = Date()
 
-        assertTrue(reporter.crashBreadcrumbs[0].date.after(beginDate))
-        assertTrue(reporter.crashBreadcrumbs[0].date.before(afterDate))
+        assertTrue(reporter.crashBreadcrumbs.elementAt(0).date.after(beginDate))
+        assertTrue(reporter.crashBreadcrumbs.elementAt(0).date.before(afterDate))
 
         val date = Date()
         reporter.recordCrashBreadcrumb(
                 Breadcrumb(testMessage, testData, testCategory, testLevel, testType, date)
         )
-        assertTrue(reporter.crashBreadcrumbs[1].date.compareTo(date) == 0)
+        assertTrue(reporter.crashBreadcrumbs.elementAt(1).date.compareTo(date) == 0)
+    }
+
+    @Test
+    fun `Reporter stores the correct priority of breadcrumbs`() {
+        val testMessage = "test_Message"
+        val testData = hashMapOf("1" to "one", "2" to "two")
+        val testCategory = "testing_category"
+        val testType = Breadcrumb.Type.USER
+
+        val service = object : CrashReporterService {
+            override fun report(crash: Crash.UncaughtExceptionCrash) {
+            }
+
+            override fun report(crash: Crash.NativeCodeCrash) {
+            }
+        }
```

Comment:
nit: If you only need an empty implementation then a mock may be sufficient?

```kotlin
al service : CrashReporterService = mock()
```

Reply:
I'll try it.

Reply:
Works great thanks!